workflows/alignment-counts-workflow.nf

```diff
@@ -71,79 +75,78 @@ process short_read_alignment {
 
 // COMPUTE ALIGNMENT STATS FOR ALL STATS
 process sam_to_stats {
-
-    //publishDir "${params.phip_data_dir}/stats"
-    label 'samtools'
-
     input:
-        tuple val(sample_id), path(sam_file)
-    
+    tuple val(sample_id), path(sam_file)
     output:
-        path "${sample_id}.stats"
-    
+    path "${sample_id}.stats"
     shell:
     template "sam_to_stats.sh"
-
 }
 
 
 // COMPUTE COUNTS FOR ALL SAMPLES
 process sam_to_counts {
-    
-    //publishDir "${params.phip_data_dir}/counts/"
-    label 'samtools'
-
+    //label 'samtools'
+    publishDir 'counts'
     input:
-        tuple val(sample_id), path(sam_file)
-
+    tuple val(sample_id), path(sam_file)
     output:
-        path "${sample_id}.counts"
-
+    path "${sample_id}.counts"
     shell:
     template "sam_to_counts.sh"
 }
 
 
 // COLLECT AND MERGE ALL 
 process collect_phip_data {
-    
-    publishDir "${params.phip_data_dir}/", mode: 'copy', overwrite: true
-    label 'phippery'
-
     input:
-        path all_counts_files
-        path all_alignment_stats
-        path sample_table 
-        path peptide_table 
-
+    path all_counts_files
+    path all_alignment_stats
+    path sample_table 
+    path peptide_table 
     output:
-        path "${params.dataset_prefix}.phip"
+    path "${params.dataset_prefix}.phip"
 
     shell:
     template "collect_phip_data.sh"
 }
 
+process replicate_counts {
+    input:
+    path ds
+    output:
+    path ds
```

Comment:
Note that modifying a file in-place will have unexpected effects when re-running a workflow locally

Reply:
Do you mean because the inputs and get cached? Oh I guess because file references are passed around so the next time this is run the input will have been modified already ... Is this the problem?

Reply:
The file will be modified in place, which just means that if the user ends up re-running the workflow with the `-resume` flag, the input to this process could end up being the file which was _output_ by this process in the previous run. I'm not sure if this is actually a problem, but I just wanted to let you know that was how the caching behavior in Nextflow worked with this pattern.